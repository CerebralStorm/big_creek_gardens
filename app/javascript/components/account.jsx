import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'

class Account extends React.Component {
  render() {
    return (
      <div className='container section'>
        <h1>My Account</h1>
      </div>
    );
  }
}

export default Account;