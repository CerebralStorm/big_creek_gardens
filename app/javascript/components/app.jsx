import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import Header from './header'
import Footer from './footer'
import Index from './index'

import { library } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faShoppingCart } from '@fortawesome/free-solid-svg-icons'

library.add(faShoppingCart)

class App extends React.Component {
  render() {
    return (
      <div>
        <Header />
        {this.props.children}
        <Footer />
      </div>
    );
  }
}

export default App;