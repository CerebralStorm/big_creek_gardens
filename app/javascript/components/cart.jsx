import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { loadCart } from '../actions/cart'
import { Link } from 'react-router-dom'

import CartItem from './cart/cart_item'
import CheckoutForm from './cart/checkout_form'
import { Elements, StripeProvider } from 'react-stripe-elements';

class Cart extends React.Component {
  componentDidMount() {
    this.props.dispatch(loadCart())
  }

  render() {
    return (
      <div className="container cart-container">
        <div className="card shopping-cart">
          <div className="card-header bg-dark text-light">
            <i className="fa fa-shopping-cart" aria-hidden="true"></i>
            Shipping cart
            <Link to={'/shop'} className="btn btn-outline-success btn-sm pull-right">
              Continue Shopping
            </Link>
            <div className="clearfix"></div>
          </div>
          <div class="card-body">
            {Object.values(this.props.cart).map(cartItem => (
              <div>
                <CartItem key={cartItem.product.id} cartItem={cartItem} />
                <hr />
              </div>
            ))}
          </div>
        </div>
        <div class="card-footer">
          <div class="coupon col-md-5 col-sm-5 no-padding-left pull-left">
            <div class="row">
              <div class="col-6">
                <input type="text" class="form-control" placeholder="coupon code" />
              </div>
              <div class="col-6">
                <input type="submit" class="btn btn-default" value="Use coupon" />
              </div>
            </div>
          </div>
          <div class="pull-right" style={{margin: '10px'}}>
            <a href="" class="btn btn-success pull-right">Checkout</a>
            <div class="pull-right total-price" style={{margin: '5px'}}>
              <strong>Total price: <b>$50.00</b></strong>
            </div>
          </div>
          <div className='clearfix'></div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    currentUser: state.currentUser,
    cart: state.cart
  }
}

export default connect(mapStateToProps)(Cart)