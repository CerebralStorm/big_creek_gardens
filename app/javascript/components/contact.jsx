import React from 'react'
import MessageApi from '../api/message_api'
import { connect } from 'react-redux'
import { showFlashMessage } from '../actions/flash'

class Contact extends React.Component {
  constructor(props) {
    super(props);
    this.submitMessage = this.submitMessage.bind(this)
    this.handleChange = this.handleChange.bind(this)
    this.state = {
      name: '',
      email: '',
      message: ''
    }
  }

  handleChange(event) {
    let state = this.state
    state[event.target.name] = event.target.value
    this.setState(state)
  }

  submitMessage(event) {
    event.preventDefault()
    let message = {
      name: this.state.name,
      email: this.state.email,
      message: this.state.message
    }
    MessageApi.createMessage({message: message}).then(() => {
      this.props.dispatch(showFlashMessage('Thank you for your message', 'success'))
      this.setState({
        name: '',
        email: '',
        message: ''
      })
    })
  }

  render() {
    return (
      <div className="container mt-3 mb-3">
        <h2>Contact Us</h2>
        <p>Feel free to shout us by filling the contact form or visiting our social network sites like Fackebook,Whatsapp,Twitter.</p>
        <div className="row">
          <div className="col-md-8 col-md-offset-2">
            <form className="form-horizontal">
              <div className="form-group">
                <label htmlFor="contact-name">Name</label>
                <input name="name" type="text" className="form-control" id="contact-name" onChange={this.handleChange} value={this.state.name} />
              </div>
              <div className="form-group">
                <label htmlFor="contact-email">Email</label>
                <input name="email" type="email" className="form-control" id="contact-email" onChange={this.handleChange} value={this.state.email} />
              </div>
              <div className="form-group ">
                <label htmlFor="contact-text">Your Message</label>
               <textarea name="message" className="form-control" id="contact-text" onChange={this.handleChange} value={this.state.message}></textarea>
              </div>
              <button type="submit" className="btn btn-default" onClick={this.submitMessage} >Send Message</button>
            </form>

            <hr />
            <h3>Our Social Sites</h3>
            <ul className="list-inline banner-social-buttons">
              <li><a href="#" className="btn btn-default btn-lg"><i className="fa fa-twitter"> <span className="network-name">Twitter</span></i></a></li>
              <li><a href="#" className="btn btn-default btn-lg"><i className="fa fa-facebook"> <span className="network-name">Facebook</span></i></a></li>
              <li><a href="#" className="btn btn-default btn-lg"><i className="fa fa-youtube-play"> <span className="network-name">Youtube</span></i></a></li>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    currentUser: state.currentUser,
    flash: state.flash
  }
}

export default connect(mapStateToProps)(Contact)
export { Contact };