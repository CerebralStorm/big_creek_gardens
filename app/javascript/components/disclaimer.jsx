import React from 'react'
import Modal from 'react-bootstrap4-modal';

class Disclaimer extends React.Component {
  constructor(props) {
    super(props);
    this.toggleModal = this.toggleModal.bind(this)
    this.state = {
      show: false
    }
  }

  toggleModal(event) {
    this.setState({show: !this.state.show})
  }

  render() {
    return (
      <span className='link nav-link'>
        <span onClick={this.toggleModal}>Disclaimer</span>
        <Modal visible={this.state.show} onClickBackdrop={this.toggleModal}>
           <div className="modal-header">
             <h5 className="modal-title">Disclaimer</h5>
           </div>
           <div className="modal-body">
             <p>Notice to law enforcement authorities: What is contained on this page might look like marijuana, but it is actually legal industrial hemp flower grown in compliance with the Agricultural Act of 2014, 7 U.S.C. §5940 (2014 Farm Bill), pursuant to pilot programs lawfully established under 7 U.S.C. §5940 in conformance with California Food and Agricultural Code, Division 24, §81001, et seq., and Nevada Revised Statutes §557.010, et seq. California’s and Nevada’s respective Departments of Agriculture have certified, registered, and regulated the growing of our industrial hemp in conformance with federal and state laws, rules, and regulations. Federal law, in particular, 7 U.S.C. §5940(b)(1), entitled “Legitimacy of Industrial Hemp Research,” encourages growing, cultivating, and marketing industrial hemp products pursuant to these pilot programs, and by the rights and protections thereby established, CBDHemp.direct proudly offers this legal industrial hemp flower high in cannabidiol (CBD) in the United States.</p>
             <p>According to 7 U.S.C. §5940, the term “industrial hemp” means the plant Cannabis sativa L. and any part of such plant, whether growing or not, with a Delta-9 tetrahydrocannabinol (Delta-9 THC) concentration of not more than 0.3% on a dry weight basis. Only the Delta-9 THC level is relevant, not THC-A. As shown by the enclosed Certificate of Analysis, this hemp flower has a Delta-9 THC level on a dry weight basis equal to 0%, well below the 0.3% maximum level and, therefore, this flower is hemp, not marijuana, and is perfectly legal to possess and sell. This right applies in any state pursuant to the Full Faith and Credit Clause, Article VI, Section 1 of the Constitution, the Supremacy Clause, Article VI, Section 2 of the Constitution, and the Equal Protection Clause, Section 1 of the Fourteenth Amendment.</p>
           </div>
           <div className="modal-footer">
             <button type="button" className="btn btn-secondary" onClick={this.toggleModal}>
               Close
             </button>
           </div>
        </Modal>
      </span>
    );
  }
}

export default Disclaimer
export { Disclaimer };