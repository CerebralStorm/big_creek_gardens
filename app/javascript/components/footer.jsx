import React from 'react'
import Disclaimer from './disclaimer'
import { Link } from 'react-router-dom'

class Footer extends React.Component {
  render() {
    return (
      <footer>
        <div className="container">
          <div className="row section">
            <div className="footer-widget col-md-3 col-xs-12 wow fadeIn">
              <h3 className="small-title">
                ABOUT US
              </h3>
              <p>
                Etiam ornare condimentum massa et scelerisque. Mauris nibh ipsum, laoreet at venenatis ac, rutrum sed risus,
              </p>
              <p>Aliquam magna nibh, mattis a urna nec. Semper venenatis magna.</p>
              <div className="social-footer">
                <a href="#"><i className="fa fa-facebook icon-round"></i></a>
                <a href="#"><i className="fa fa-twitter icon-round"></i></a>
                <a href="#"><i className="fa fa-linkedin icon-round"></i></a>
                <a href="#"><i className="fa fa-google-plus icon-round"></i></a>
              </div>
            </div>

            <div className="footer-widget col-md-3 col-xs-12 wow fadeIn" data-wow-delay=".2s">
              <h3 className="small-title">
                TWITTER
              </h3>
              <ul className="recent-tweets">
                <li className="tweet">
                  My <a href="#">@Quora</a>
                  <span className="tweet-text">
                    answer to What's the best FAQ plugin for WordPress?
                  </span>
                  <a href="#">qr.ae/RFTbIGa</a>
                  <span className="tweet-date">
                    August 21, 2015 10:29pm
                  </span>
                </li>
                <li className="tweet">
                  WPB Advanced FAQ | Probably The Best WordPress FAQ Plugin <a href="#">wpbean.com/wpb-advanced-faq-pr…</a>
                  <span className="tweet-text">
                    via
                  </span>
                  <a href="#">@wpbean</a>
                  <span className="tweet-date">
                    August 19, 2015 8:49 am
                  </span>
                </li>
              </ul>
            </div>

            <div className="footer-widget col-md-3 col-xs-12 wow fadeIn" data-wow-delay=".5s">
              <h3 className="small-title">
                GALLERY
              </h3>
              <div className="plain-flicker-gallery">
                <a href="#" title="Pan Masala"><img src="flicker/img1.jpg" alt="" /></a>
                <a href="#" title="Sports Template for Joomla"><img src="flicker/img2.jpg" alt="" /></a>
                <a href="" title="Apple Keyboard"><img src="flicker/img3.jpg" alt="" /></a>
                <a href="" title="Hard Working"><img src="flicker/img4.jpg" alt="" /></a>
                <a href="" title="Smile"><img src="flicker/img5.jpg" alt="" /></a>
                <a href="" title="Puzzle"><img src="flicker/img6.jpg" alt="" /></a>
              </div>
            </div>

            <div className="footer-widget col-md-3 col-xs-12 wow fadeIn" data-wow-delay=".8s">
              <h3 className="small-title">
                SUBSCRIBE US
              </h3>
              <div className="contact-us">
                <p>Tellus aliquam at. Pellentesque gravida vel eros et pretium</p>
                <form>
                <div className="form-group">
                  <input type="text" className="form-control" id="exampleInputName2" placeholder="Enter your name" />
                </div>
                <div className="form-group">
                  <input type="email" className="form-control" id="exampleInputEmail2" placeholder="Enter your email" />
                </div>
                <button type="submit" className="btn btn-common">Submit</button>
              </form>
              </div>
            </div>
          </div>
        </div>

        <div id="copyright">
          <div className="container">
            <div className="row">
              <div className="col-md-6 col-sm-6">
                <p className="copyright-text">
                  ©  2018 Area 51 Hemp. All right reserved.
                </p>
              </div>
              <div className="col-md-6  col-sm-6">
                <ul className="nav nav-inline pull-xs-right">
                  <li className="nav-item">
                    <Link to={'/'} className="nav-link">
                      Home
                    </Link>
                  </li>
                  <li className="nav-item">
                    <Disclaimer className="nav-link" />
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="#">Privacy Policy</a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="#">Terms of services</a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </footer>
    );
  }
}

export { Footer };
export default Footer;