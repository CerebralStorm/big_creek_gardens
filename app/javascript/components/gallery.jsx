import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'

class Gallery extends React.Component {
  render() {
    return (
      <div>
        <h1>Gallery</h1>
      </div>
    );
  }
}

export default Gallery;