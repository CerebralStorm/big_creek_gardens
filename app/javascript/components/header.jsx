import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import logo from '../assets/images/logo-alpha.png'
import { Link } from 'react-router-dom'

const Header = props => (
  <header>
    <nav className="navbar fixed-top navbar-expand-lg">
      <div className='container'>
        <Link to={'/'} className='navbar-brand'>
          <img src={logo} width="100" />
        </Link>
        <button className="navbar-toggler hidden-md-up pull-xs-right" data-target="#navbarSupportedContent" data-toggle="collapse" type="button">
          â˜°
        </button>

        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <div className="menu ml-auto">
            <ul className="navbar-nav ml-auto">
              <li className="nav-item active">
                <Link to={'/'} className="nav-link">
                  Home
                </Link>
              </li>
              <li className="nav-item">
                <Link to={'/shop'} className="nav-link">
                  Shop
                </Link>
              </li>
              <li className="nav-item">
                <Link to={'/gallery'} className="nav-link">
                  Gallery
                </Link>
              </li>
              <li className="nav-item">
                <Link to={'/contact'} className="nav-link">
                  Contact
                </Link>
              </li>
              <li className="nav-item">
                <a href='/users/sign_in' className="nav-link">
                  Sign In
                </a>
              </li>
              <li className="nav-item">
                <a href='/users/sign_up' className="nav-link">
                  Sign Up
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </nav>
  </header>
)

export default Header;

