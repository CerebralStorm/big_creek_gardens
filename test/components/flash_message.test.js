import React from 'react'
import { FlashMessage } from '../../app/javascript/components/flash_message'

let flash = {
  message: 'this is a test',
  messageType: 'success'
}

test('render a FlashMessage', () => {
  const wrapper = shallow(<FlashMessage flash={flash} />);
  expect(wrapper).toMatchSnapshot();
});