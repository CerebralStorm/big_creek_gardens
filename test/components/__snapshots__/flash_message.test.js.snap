// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a FlashMessage 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <div
    className="alert alert-success"
    role="alert"
  >
    this is a test
    <button
      aria-label="Close"
      className="close"
      data-dismiss="alert"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
    </button>
  </div>
</div>
`;
